### PR TITLE
nix: clean up files, use good practices

### PR DESCRIPTION
**Pull request by RGBCube**
- **nix: clean up files**
- **nix: update docs to reflect the new setup**

### CODE REVIEW

**Comment by RGBCube** [comment]
I also moved the new nix command to the top, as it should be preferred.

File: Ladybird/default.nix [hunk 1]
```diff
@@ -1,12 +1,7 @@
...
-{
-  stdenv = pkgs.gcc13Stdenv;
-}
-{
-  name = "ladybird";
+{ pkgs ? import <nixpkgs> { } }: with pkgs;
 
-  nativeBuildInputs = with pkgs; [
+mkShell.override { stdenv = gcc13Stdenv; } {
+  packages = [
```
Comment (MacDue) [inline Ladybird/default.nix:4]: I think `qt6.qtsvg` can be removed from these dependencies (the Qt chrome has not needed that for a while).

File: Toolchain/default.nix [hunk 2]
```diff
@@ -21,9 +19,6 @@ stdenv.mkDerivation {
     rsync
     texinfo
     unzip
-  ];
-
-  buildInputs = [
     e2fsprogs
     fuse2fs
```
Comment (ADKaster) [inline Toolchain/default.nix:23]: there's duplicates here for e2fsprogs and fuse2fs

Comment (RGBCube) [reply Toolchain/default.nix:23]: Which one is needed? I'll edit the pr

Comment (ADKaster) [reply Toolchain/default.nix:23]: oh, I meant that both of those packages are duplicates with further up in the package list (they used to be both in nativeBuildInputs and buildInputs)

**Comment by RGBCube** [comment]
Should be good to go then, I've applied the changes requested.

**Review by ADKaster — APPROVED** [review]

**Comment by networkException** [comment]
hm I don't like how we're declaring this as "good practice" that are *in fact* (in practice) considered experimental in Nix. The previous description also stated the difference between the v2 and v3 cli, this does not.

**Comment by RGBCube** [comment]
> hm I don't like how we're declaring this as "good practice" that are _in fact_ (in practice) considered experimental in Nix. The previous description also stated the difference between the v2 and v3 cli, this does not.

The new description mentions that the v2 cli is old and will use the host nixpkgs, I think that's pretty descriptive enough?

Defining flakes as good practice is controversial, but it's the general way the community is going towards as it is more reproducible. Companies also prioritize flake support. (See DetSys, Hercules CI, Garnix and a few more which are all aimed towards flakes)

**Pull request merged**